.github/workflows/continuous-integration.yml

```diff
@@ -21,28 +24,32 @@ jobs:
         ports:
           - 5432:5432
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
-      - run: npm install
-      - run: npm test
+      - name: Update NPM
+        if: matrix.node-version <= 14
+        run: npm install -g npm@latest-7
```

Comment:
From my testing, updating NPM may not be necessary.

Reply:
Github actions is currently overwriting the package-lock file during testing. If you change and do npm ci will fail

```
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
```

Reply:
The documentation says that NPM v7 supports package-lock v1, v2 and v3 but Node v12 and v14 come with lower NPM versions

Reply:
AFAIK, the warning is just a warning as they didn't update NPM <=v6 to explicitly recognise v2 lockfiles, but v2 lockfiles were designed to be backwards-compatible nonetheless.

From https://blog.npmjs.org/post/626173315965468672/npm-v7-series-beta-release-and-semver-major.html:

> package-lock.json is updated to a newer format, using "lockfileVersion": 2. This format is backwards-compatible with npm CLI versions using "lockfileVersion": 1, but older npm clients will print a warning about the version mismatch

Hence my understanding of the NPM-lockfile support is:

- NPM <=v6
    - Can read v1 and v2 lockfiles, but not v3
    - Can generate v1 lockfiles, but not v2 (will gracefully downgrade) or v3 (will regenerate from scratch)
- NPM v7 & v8
    - Can read v1, v2 and v3 lockfiles
    - Can generate v2 (default) and v3 lockfiles, but not v1 (will gracefully upgrade).

> Github actions is currently overwriting the package-lock file during testing

The switch to `npm ci` solves this problem as NPM will not try to modify the lockfile. Hence we don't need to upgrade the NPM version.